src/utils/slpParser.ts

```diff
@@ -0,0 +1,114 @@
+import _ from "lodash";
+
+import { PostFrameUpdateType, GameStartType, GameEndType, Command, PreFrameUpdateType } from "./slpReader";
+import { FramesType, FrameEntryType, StatsType, Frames, PlayerIndexedType, getSinglesOpponentIndicesFromSettings } from "../stats/common";
+import { Stats } from "../stats/stats";
+
+export class SlpParser {
+    private settings: GameStartType | null = null;
+    private playerFrames: FramesType | null = null;
+    private followerFrames: FramesType | null = null;
+    private gameEnd: GameEndType | null = null;
+    private latestFrameIndex: number | null = null;
+    private statsComputer: Stats | null = null;
+    private playerIndices: PlayerIndexedType[] = [];
+
+    public constructor() {
+        this.playerFrames = {};
+        this.followerFrames = {};
+        this.statsComputer = new Stats([]);
```

Comment:
Is it worth creating a Stats object here? Is this even a valid state for a Stats object? Might be better to not even initialize? Not sure.

Reply:
Yeah it's needed because you could call `getStats()` before you process any frames. The expected return value is the default stats object. But you can't even call `getStats()` if the object isn't initialized.